public/app/features/alerting/unified/components/silences/SilencesEditor.tsx

```diff
@@ -22,145 +21,142 @@ import {
+  );
 };
 
-const getDefaultFormValues = (searchParams: URLSearchParams, silence?: Silence): SilenceFormFields => {
-  const now = new Date();
-  if (silence) {
-    const isExpired = Date.parse(silence.endsAt) < Date.now();
-    const interval = isExpired
-      ? {
-          start: now,
-          end: addDurationToDate(now, { hours: 2 }),
-        }
-      : { start: new Date(silence.startsAt), end: new Date(silence.endsAt) };
-    return {
-      id: silence.id,
-      startsAt: interval.start.toISOString(),
-      endsAt: interval.end.toISOString(),
-      comment: silence.comment,
-      createdBy: silence.createdBy,
-      duration: intervalToAbbreviatedDurationString(interval),
-      isRegex: false,
-      matchers: silence.matchers?.map(matcherToMatcherField) || [],
-      matcherName: '',
-      matcherValue: '',
-      timeZone: DefaultTimeZone,
-    };
-  } else {
-    const endsAt = addDurationToDate(now, { hours: 2 }); // Default time period is now + 2h
-    return {
-      id: '',
-      startsAt: now.toISOString(),
-      endsAt: endsAt.toISOString(),
-      comment: `created ${dateTime().format('YYYY-MM-DD HH:mm')}`,
-      createdBy: config.bootData.user.name,
-      duration: '2h',
-      isRegex: false,
-      matchers: [{ name: '', value: '', operator: MatcherOperator.equal }],
-      matcherName: '',
-      matcherValue: '',
-      timeZone: DefaultTimeZone,
-      ...defaultsFromQuery(searchParams),
-    };
-  }
+type SilencesEditorProps = {
+  formValues?: SilenceFormFields;
+  alertManagerSourceName: string;
+  onSilenceCreated?: (response: SilenceCreatedResponse) => void;
+  onCancel?: () => void;
+  ruleUid?: string;
 };
 
-export const SilencesEditor = ({ silenceId, alertManagerSourceName }: Props) => {
-  // Use a lazy query to fetch the Silence info, as we may not always require this
-  // (e.g. if creating a new one from scratch, we don't need to fetch anything)
-  const [getSilence, { data: silence, isLoading: getSilenceIsLoading, error: errorGettingExistingSilence }] =
-    alertSilencesApi.endpoints.getSilence.useLazyQuery();
+/**
+ * Base silences editor used for new silences (from both the list view and the drawer),
+ * and for editing existing silences
+ */
+export const SilencesEditor = ({
+  formValues = getDefaultSilenceFormValues(),
+  alertManagerSourceName,
+  onSilenceCreated,
+  onCancel,
+  ruleUid,
+}: SilencesEditorProps) => {
   const [createSilence, { isLoading }] = alertSilencesApi.endpoints.createSilence.useMutation();
-  const [urlSearchParams] = useURLSearchParams();
+  const [getAlertByUid, { data: alertRule }] = alertRuleApi.endpoints.getGrafanaAlertRule.useLazyQuery();
+  const formAPI = useForm({ defaultValues: formValues });
+  const styles = useStyles2(getStyles);
+
+  const { register, handleSubmit, formState, watch, setValue, clearErrors } = formAPI;
+  const duration = watch('duration');
+  const startsAt = watch('startsAt');
+  const endsAt = watch('endsAt');
```

Comment:
Could be simplified to
```suggestion
  const [duration, startsAt, endsAt] = watch(['duration', 'startsAt', 'endsAt']);
```